xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/webjar/xwiki-macro/macroEditor.js

```diff
@@ -657,16 +504,22 @@ define(
           .prop('requestNumber', requestNumber);
 
         // Load the macro descriptor
-        macroService.getMacroDescriptor(macroId, sourceDocumentReference)
-          .done(maybeCreateMacroEditor.bind(macroEditor, requestNumber, macroCall))
-          .fail(maybeShowError.bind(macroEditor, requestNumber, 'descriptorRequestFailed'));
+        // FIXME: only perform this call if needed
+        macroService.getMacroParametersFromHTML(macroId, widgetHtml)
+            // TODO: this should be probably improved with better $.then construction.
+          .done(function (parameters) {
+                macroService.getMacroDescriptor(macroId, sourceDocumentReference)
+                    .done(function (descriptor) {
+                      maybeCreateMacroEditor.bind(macroEditor, requestNumber, macroCall, descriptor, parameters)();
```

Comment:
@mflorea this is particularly ugly but I failed to make it cleaner, I'm taking suggestion.